ruma-api-macros/src/util.rs

```diff
@@ -257,3 +257,11 @@ pub(crate) fn req_res_name_value<T>(
     *header = Some(value);
     Ok(field_kind)
 }
+
+fn is_ascii_printable(b: &u8) -> bool {
+    b.wrapping_sub(b' ') < 0x5F
```

Comment:
That's a weird check, I can't tell if it is correct. Does this do the same thing as `(0x20..=0x7E).contains(b)`?

Reply:
That works the same I changed it to the above thats cool you can use the range like that!

Is space " " an invalid character for paths?

Reply:
Now that you mention it, that is considered printable, huh? Yeah, let's forbid it for now. It would have to be percent-encoded at the very least, but I don't see Matrix ever adding endpoints with spaces in their paths :D